homeassistant/__main__.py

```diff
@@ -256,10 +258,29 @@ async def setup_and_run_hass(config_dir: str, args: argparse.Namespace) -> int:
     if hass is None:
         return 1
 
-    if args.open_ui and hass.config.api is not None:
+    if args.open_ui:
         import webbrowser  # pylint: disable=import-outside-toplevel
+        from homeassistant.helpers import (
+            network,
+        )  # pylint: disable=import-outside-toplevel
+
+        url: Optional[str] = None
+        try:
+            url = network.async_get_url(hass)
+        except network.NoURLAvailableError:
+            pass
+
+        if url is None and hass.config.api is not None:
+            # Fallback to localhost
+            scheme = "https" if hass.config.api.use_ssl else "http"
+            url = str(
+                yarl.URL.build(
+                    scheme=scheme, host="127.0.0.1", port=hass.config.api.port
+                )
+            )
 
-        hass.add_job(webbrowser.open, hass.config.api.base_url)
+        if url is not None:
+            hass.add_job(webbrowser.open, url)
```

Comment:
`webbrowser.open()` opens the browser on the computer that is running Home Assistant. It was an early on feature to help people get started. I think that we can just default to localhost and `hass.config.api.port` here .